sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Processor/EventProcessorClientTests.cs

```diff
@@ -700,113 +674,1998 @@ public async Task IsRunningReturnsTrueWhileStopProcessingAsyncIsNotCalled()
+        ///   after <see cref="EventProcessorClient.StopProcessingAsync" /> is called.
+        /// </summary>
+        ///
+        [Test]
+        public async Task StopProcessingAsyncStopsLoadbalancer()
+        {
+            var mockLoadbalancer = new Mock<PartitionLoadBalancer>();
+            mockLoadbalancer.SetupAllProperties();
+            mockLoadbalancer.Object.LoadBalanceInterval = TimeSpan.FromSeconds(1);
+            Func<EventHubConnection> connectionFactory = () => new MockConnection();
+            var connection = connectionFactory();
+            var storageManager = new MockCheckPointStorage((s) => Console.WriteLine(s));
+            var processor = new MockEventProcessorClient(
+                storageManager,
+                connectionFactory: connectionFactory,
+                loadBalancer: mockLoadbalancer.Object);
+
+            await processor.StartProcessingAsync();
+            await processor.StopProcessingAsync();
+
+            // Stopping the processor should stop the PartitionLoadBalancer.
+
+            mockLoadbalancer.Verify(m => m.RelinquishOwnershipAsync(It.IsAny<CancellationToken>()));
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="EventProcessorClient.StopProcessingAsync" />
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        public async Task StopProcessingAsyncStopsProcessingForEveryPartition()
+        {
+            var maximumWaitTimeInMilli = 100;
+            var processorOptions = new EventProcessorClientOptions { MaximumWaitTime = TimeSpan.FromMilliseconds(maximumWaitTimeInMilli) };
+            var mockConsumer = new Mock<EventHubConsumerClient>("consumerGroup", Mock.Of<EventHubConnection>(), default);
+            var mockProcessor = new InjectableEventSourceProcessorMock(new MockCheckPointStorage(), "consumerGroup", "namespace", "eventHub", Mock.Of<Func<EventHubConnection>>(), processorOptions, mockConsumer.Object);
+
+            var partitionIds = new[] { "0", "1" };
+
+            mockConsumer
+                .Setup(consumer => consumer.GetPartitionIdsAsync(It.IsAny<CancellationToken>()))
+                .Returns(Task.FromResult(partitionIds));
+
+            var completionSource = new TaskCompletionSource<bool>();
+            var partitionsBeingProcessed = 0;
+
+            // Use the ReadEventsFromPartitionAsync method to notify when all partitions have started being processed.
+
+            mockConsumer
+                .Setup(consumer => consumer.ReadEventsFromPartitionAsync(
+                    It.IsAny<string>(),
+                    It.IsAny<EventPosition>(),
+                    It.IsAny<ReadEventOptions>(),
+                    It.IsAny<CancellationToken>()))
+                .Callback(() =>
+                {
+                    if (Interlocked.Increment(ref partitionsBeingProcessed) == partitionIds.Length)
```

Comment:
Should we consider `>=` for safety here?

Reply:
It won't hurt. I'll update the `SetResult` to `TrySetResult` as well just in case.